File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        
        if len(s) != len(t):
            return False
        
        i = 0
        s1, s2 = {}, {}

        while i < len(s):
            c1, c2 = s[i], t[i]
            s1[c1] = s1.get(c1, 0) + 1
            s2[c2] = s2.get(c2, 0) + 1
            i += 1

        if len(s1.keys()) != len(s2.keys()) :
            return False

        return s1 == s2

File: Data Structures & Algorithms/subsets/submission-1.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        final List<List<Integer>> result = new ArrayList<>();
        subset(0, result, new ArrayList<>(), nums);
        return result;
    }

    private static void subset(
        int i, List<List<Integer>> result, List<Integer> current, int[] nums) {
        if (i >= nums.length) {
            result.add(new ArrayList<>(current));
            return;
        }

        current.add(nums[i]);
        subset(i + 1, result, current, nums);

        current.remove(current.size()-1);
        subset(i + 1, result, current, nums);
    }
}
